src/lib/operators/sort.cpp

```diff
@@ -33,19 +20,77 @@ const std::string& Sort::name() const {
 std::shared_ptr<AbstractOperator> Sort::_on_deep_copy(
     const std::shared_ptr<AbstractOperator>& copied_input_left,
     const std::shared_ptr<AbstractOperator>& copied_input_right) const {
-  return std::make_shared<Sort>(copied_input_left, _column_id, _order_by_mode, _output_chunk_size);
+  return std::make_shared<Sort>(copied_input_left, _sort_definitions, _output_chunk_size);
 }
 
 void Sort::_on_set_parameters(const std::unordered_map<ParameterID, AllTypeVariant>& parameters) {}
 
 std::shared_ptr<const Table> Sort::_on_execute() {
-  _impl = make_unique_by_data_type<AbstractReadOnlyOperatorImpl, SortImpl>(
-      input_table_left()->column_data_type(_column_id), input_table_left(), _column_id, _order_by_mode,
-      _output_chunk_size);
-  return _impl->_on_execute();
+  _sort_definition_data_types.resize(_sort_definitions.size());
+  std::transform(
+      _sort_definitions.begin(), _sort_definitions.end(), _sort_definition_data_types.begin(),
+      [this](const auto sort_definition) { return input_table_left()->column_data_type(sort_definition.column); });
+
+  std::shared_ptr<const Table> output = input_table_left();
```

Comment:
output = input? Why?

Reply:
IIRC, this is there in order to just output the input table _as is_ when the Sort operator is used with an empty `SortDefinition` struct vector as its input configuration (i. e. when there are not columns which the input should be sorted by). Does that make sense? Probably we should add a corresponding comment to the line then … 🙃

Reply:
Can you come up with a case where this would be legal? If someone creates a sort operator but does not declare what the order should be, the constructor (not _validate_sort_definitions, which is called much later) should just go up in flames.